lnwire/features.go

```diff
@@ -39,33 +39,36 @@ type FeaturesMap map[featureName]int
 type indexToFlag map[int]featureFlag
 
 const (
-	// optionalFlag represent the feature which we might be used already but
-	// it isn't required yet, and if remote peer doesn't have this feature
-	// we may turn it off without disconnecting with peer.
+	// optionalFlag represent the feature which we already have but it isn't
+	// required yet, and if remote peer doesn't have this feature we may
+	// turn it off without disconnecting with peer.
 	optionalFlag featureFlag = 2 // 0b10
 
 	// requiredFlag represent the features which is required for proper
 	// peer interaction, we disconnect with peer if it doesn't have this
 	// particular feature.
-	requiredFlag featureFlag = 3 // 0b11
+	requiredFlag featureFlag = 1 // 0b01
 
-	// flagMask is a mask which is needed to extract feature value.
+	// flagMask is a mask which is needed to extract feature flag value.
 	flagMask = 3 // 0b11
 
-	// featureBitsSize represent the size of the feature in bits. For
+	// flagBitsSize represent the size of the feature flag in bits. For
 	// more information read the init message specification.
-	featureBitsSize = 2
+	flagBitsSize = 2
+
+	// maxAllowedSize is a maximum allowed size of feature vector.
+	maxAllowedSize = 2 ^ 10 // 1024 bytes => (2 ^ 10 * 8) / 2 = 4096 features
```

Comment:
Hmm, how'd you arrive at the value for this constant? 

Within the protocol, the maximum allowed message size is `65535` bytes. Adding the overhead from the crypto protocol (the `2-byte` packet length and `16-byte` MAC), we arrive at `65569` bytes. Accounting for the overhead within the feature message to signal the type of the message, that leaves `65567` bytes for the `init` message itself. Next, we reserve `4-bytes` to encode the lengths of both the local and global feature vectors, so `65563` for the global and local features. Knocking off one byte for the sake of the calculation, that leads to a max allowed size of `32781` bytes for each feature vector, or `131124` different features. 

Does my napkin calc above make sense?